cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterConflictException.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.adapter;
+
+/**
+ * Thrown when Adapter conflict occurs. For instance, if an adapter 'stop' is requested, but the adapter is already
+ * stopped.
+ */
+public class AdapterConflictException extends Exception {
```

Comment:
Conflict is a state, the actual issue is that you cannot perform the operation expected. How about ```InvalidOperationException``` and then we can return a ```Conflict - 409``` from the HTTP layer that calls it.